File: web/src/main/java/ru/protei/portal/ui/web/client/model/event/EventBusEventAuthLogoutDone.java
package ru.protei.portal.ui.web.client.model.event;

import jsinterop.annotations.JsOverlay;

public class EventBusEventAuthLogoutDone extends EventBusEvent {
    protected EventBusEventAuthLogoutDone() {}
    public static String type = "@app-portal/auth-logout-done";

    public static native EventBusEventAuthLogoutDone create(boolean userInitiated) /*-{
        return {
            type: @ru.protei.portal.ui.web.client.model.event.EventBusEventAuthLogoutDone::type,
            source: @ru.protei.portal.ui.web.client.model.event.EventBusSource::SourcePortalUiGwt,
            payload: {
                userInitiated: userInitiated
            }
        };
    }-*/;

    @JsOverlay public final native boolean isUserInitiated() /*-{ return this.payload.userInitiated; }-*/;
}

File: webts/packages/common/common/src/js/model/eventbus/event/auth/logout/EventBusEventAuthLogoutDone.ts
import { EventBusEvent, EventSource } from "@protei-libs/eventbus"
import { EventBusSourcePortalUi } from "../../../source/EventBusSource"

export const EventBusEventAuthLogoutDoneType = "@app-portal/auth-logout-done"

export type EventBusEventAuthLogoutDone = ReturnType<typeof eventBusEventAuthLogoutDone>

export type EventBusEventAuthLogoutDonePayload = {
  userInitiated: boolean
}

export const eventBusEventAuthLogoutDone = (
  payload: EventBusEventAuthLogoutDonePayload,
  options?: { source?: EventSource },
): EventBusEvent<EventBusEventAuthLogoutDonePayload> =>
  <const>{
    type: EventBusEventAuthLogoutDoneType,
    source: options?.source ?? EventBusSourcePortalUi,
    payload: payload,
  }
